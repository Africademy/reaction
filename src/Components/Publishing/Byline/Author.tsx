import styled from "styled-components"
import { Sans } from "@artsy/palette"
import React from "react"
import { pMedia } from "../../Helpers"
import { getAuthorByline } from "../Constants"
import { BylineLayout } from "../Typings"

interface AuthorProps {
  authors?: any
  color?: string
  layout?: BylineLayout
  size?:
    | "1"
    | "2"
    | "3"
    | "3t"
    | "4"
    | "4t"
    | "5"
    | "5t"
    | "6"
    | "8"
    | "10"
    | "12"
    | "14"
    | "16"
}

export const Author: React.SFC<AuthorProps> = props => {
  const { color, layout, size } = props
  const condensed = layout === "condensed"
  const fontSize = size ? size : condensed ? "2" : "3t"

  return (
    <Sans size={fontSize} weight="medium">
      <StyledAuthor condensed={condensed} color={color}>
        {getAuthorByline(props.authors)}
      </StyledAuthor>
    </Sans>
  )
}

Author.defaultProps = {
  color: "black",
}

export const StyledAuthor = styled.div.attrs<{
  condensed?: boolean
  color: string
}>({})`
  ${props => bullet(props.color, props.condensed)} margin: 10px 30px 0 0;
  ${props => pMedia.sm` margin: 10px 20px 0 0;`};
`

const bullet = (color, condensed = false) => {
  return `
    &:before {
      content: "";
      display: inline-block;
      width: 8px;
      height: 8px;
      border-radius: 50%;
      margin: ${condensed ? "0 5px 1px 0;" : "6px 10px 1px 0"};
      background-color: ${color};
    }
  `
}
