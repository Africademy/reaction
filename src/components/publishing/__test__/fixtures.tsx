import { Props as ImageSetPreviewProps } from "../imageset_preview"

export const Artworks = [
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
    },
    artist: {
      name: "Fernando Botero",
    },
    width: 1152,
    height: 826,
  },
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
      slug: "gary-nader",
    },
    artists: [
      {
        name: "Fernando Botero",
        slug: "fernando-botero",
      },
      {
        name: "Frida Kahlo",
        slug: "frida-kahlo",
      },
    ],
    width: 1152,
    height: 826,
  },
]

export const Articles = [
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "classic",
    vertical: null,
  },
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "standard",
    vertical: {
      name: "Art Market",
      id: "12345",
    },
  },
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "feature",
    vertical: {
      name: "Art Market",
      id: "12345",
    },
  },
]

export const Authors = [
  {
    image_url: "https://artsy-media-uploads.s3.amazonaws.com/9vcX6FR21rKHatmvJ8K0sg%2FAbigail.jpg",
    name: "Abigail Cain",
    bio: "[Abigail Cain](https://artsy.net/abigail) is an Art Market Editor at Artsy",
    twitter_handle: "abigailcain",
  },
  {
    image_url: "http://files.artsy.net/images/molly_01.jpg",
    name: "Molly Gottschalk",
    bio:
      "[Molly Gottschalk](https://artsy.net/molly) is a Features Producer at Artsy. She is a photographer and lover of cats.",
    twitter_handle: "mollygottschalk",
  },
  {
    name: "Halley Johnson",
    bio: "[Halley Johnson](https://artsy.net/molly) is an Writer Guru at Artsy. She enjoys speaking Georgian.",
    twitter_handle: "halleyjohnson",
  },
]

export const Embeds = [
  {
    mobile_height: 1300,
    height: 1000,
    url: "http://files.artsy.net/documents/1parrasch.html",
    layout: "overflow",
    type: "embed",
  },
]

export const HeroSections = [
  {
    type: "text",
    url: "https://artsy-media-uploads.s3.amazonaws.com/YqTtwB7AWqKD95NGItwjJg%2FRachel_Rossin_portrait_2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "split",
    url: "https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "fullscreen",
    url: "https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "split",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "fullscreen",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "text",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
]

export const Images = [
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
      slug: "gary-nader",
    },
    artists: [
      {
        name: "Fernando Botero",
        slug: "fernando-botero",
      },
    ],
    artist: {
      name: "Fernando Botero",
      slug: "fernando-botero",
    },
    width: 1152,
    height: 826,
  },
  {
    url: "https://artsy-media-uploads.s3.amazonaws.com/co8j2xq40ROMyBrJQm_4eQ%2FDafenOilPaintingVillage_AK03.jpg",
    type: "image",
    width: 900,
    height: 1200,
    caption: "<p>Photo by Adam Kuehl for Artsy.</p>",
  },
  {
    url: "https://d32dm0rphc51dk.cloudfront.net/CpHY-DRr7KW0HGXLslCXHw/larger.jpg",
    type: "image",
    width: 816,
    height: 1024,
    caption: "<p>Photo by Adam Kuehl for Artsy. Image courtesy of the Guggenheim Museum.</p>",
  },
]

export const ImageSetFull: ImageSetPreviewProps["section"] = {
  type: "image_set",
  layout: "full",
  title: "The Work of Bruce M. Sherman",
  images: Images,
}

export const ImageSetMini: ImageSetPreviewProps["section"] = {
  type: "image_set",
  title: "The Work of Bruce M. Sherman",
  images: Images,
}

export const ImageSetFullSansTitle: ImageSetPreviewProps["section"] = {
  type: "image_set",
  layout: "full",
  images: Images,
}

export const ImageSetMiniSansTitle: ImageSetPreviewProps["section"] = {
  type: "image_set",
  images: Images,
}
export const Videos = [
  {
    url: "https://www.youtube.com/watch?v=PXi7Kjlsz9A",
    caption: "<p>What motivates patrons to fund artists’ wildest dreams?</p>",
    cover_image_url: "https://artsy-media-uploads.s3.amazonaws.com/IB6epb5L_l0rm9btaDsY7Q%2F14183_MDP_Evening_240.jpg",
  },
  {
    url: "https://vimeo.com/191988155",
    caption:
      "<p>2016 was a memorable year for the world, and art along with it. Powered by data culled from Artsy as well as UBS’s Planet Art app, “The Year in Art 2016” will explore how the creative community responded to the cultural shifts and tribulations this year has seen—from the destruction of Palmyra to the proliferation of Virtual Reality to the U.S. election.</p>",
    cover_image_url: "https://artsy-media-uploads.s3.amazonaws.com/ditbyaUgdcl6mHin07TfKA%2FMassimilianoGioni_0581.jpg",
  },
  {
    url: "https://vimeo.com/191988155",
  },
]
