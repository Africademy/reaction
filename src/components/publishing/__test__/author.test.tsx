import * as React from "react"
import * as renderer from "react-test-renderer"

import Author from "../author"
import { Authors } from "./fixtures"

it("renders properly", () => {
  const artwork = renderer.create(<Author author={Authors[0]} />).toJSON()
  expect(artwork).toMatchSnapshot()
})
