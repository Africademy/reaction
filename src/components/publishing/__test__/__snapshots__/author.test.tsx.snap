// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="file__AuthorContainer-s1asr4y9-0 emAQlw"
>
  <div
    className="sc-bdVaJa hwyJgp"
    src="https://artsy-media-uploads.s3.amazonaws.com/9vcX6FR21rKHatmvJ8K0sg%2FAbigail.jpg"
  />
  <div
    className="file__AuthorInfo-s1asr4y9-1 grPqae"
  >
    <span>
      <a
        href="https://artsy.net/abigail"
        title={undefined}
      >
        Abigail Cain
      </a>
       is an Art Market Editor at Artsy
    </span>
    <span
      className="file__Twitter-s1asr4y9-2 kxuiUB"
    >
      <a
        className="file__TwitterHandle-s1asr4y9-3 flWIbf"
        href="http://twitter.com/abigailcain"
      >
        <div
          className="file-s1ekx40u-0 lhOxy"
          style={undefined}
        >
          î˜•
        </div>
        @abigailcain
      </a>
    </span>
  </div>
</div>
`;
